libraries/animation/src/AnimNodeLoader.cpp

```diff
@@ -564,9 +567,53 @@ AnimNode::Pointer loadInverseKinematicsNode(const QJsonObject& jsonObj, const QS
     READ_OPTIONAL_STRING(solutionSourceVar, jsonObj);
 
     if (!solutionSourceVar.isEmpty()) {
-        qCWarning(animation) << "SolutionSource(v): " << solutionSourceVar; //node->setSolutionSourceVar(solutionSourceVar);
+        node->setSolutionSourceVar(solutionSourceVar);
+    }
+
+    return node;
+}
+
+AnimNode::Pointer loadDeepMotionNode(const QJsonObject& jsonObj, const QString& id, const QUrl& jsonUrl) {
+    auto node = std::make_shared<DeepMotionNode>(id);
+
+    auto targetsValue = jsonObj.value("targets");
+    if (!targetsValue.isArray()) {
+        qCCritical(animation) << "AnimNodeLoader, bad array \"targets\" in deepMotion node, id =" << id << ", url =" << jsonUrl.toDisplayString();
+        return nullptr;
     }
 
+    auto targetsArray = targetsValue.toArray();
+    for (const auto& targetValue : targetsArray) {
+        if (!targetValue.isObject()) {
+            qCCritical(animation) << "AnimNodeLoader, bad state object in \"targets\", id =" << id << ", url =" << jsonUrl.toDisplayString();
+            return nullptr;
+        }
+        auto targetObj = targetValue.toObject();
+
+        READ_STRING(jointName, targetObj, id, jsonUrl, nullptr);
```

Comment:
What's the meaning of last argument here? 
I guess it's a kind of "pattern" to write these reads like this, I'm just wondering.